fashion-example-query/tests/test_fashion_example_search.py

```diff
@@ -0,0 +1,182 @@
+
+
+def index_generator(num_doc: int, target: dict):
+    for j in range(num_doc):
+        label_int = target['index-labels']['data'][j][0]
+        if label_int < 3:  # We are using only 3 categories, no need to index the rest
+            with Document() as d:
+                d.content = target['index']['data'][j]
+                category = get_mapped_label(label_int)
+                d.tags['label'] = category
+            yield d
+
+
+def query_generator(num_doc: int, target: dict):
+    for j in range(num_doc):
+        label_int = target['query-labels']['data'][j][0]
+        category = get_mapped_label(label_int)
+
+        if category == 'Pullover':
+            d = Document(content=(target['query']['data'][j]))
+            d.tags['label'] = category
+            yield d
+
+
+def config(task):
+    parallel = 2 if task == 'index' else 1
+    shards = 1
+    os.environ['JINA_RESOURCE_DIR'] = resource_filename('jina', 'resources')
+    os.environ['JINA_SHARDS'] = str(shards)
+    os.environ['JINA_PARALLEL'] = str(parallel)
+    os.environ['JINA_WORKDIR'] = './workspace'
+    os.makedirs(os.environ['JINA_WORKDIR'], exist_ok=True)
+    os.environ['JINA_PORT'] = os.environ.get('JINA_PORT', str(45683))
+
+
+def index(num_doc, target: dict):
+    f = Flow.load_config('flows/index.yml')
+
+    with f:
+        f.index(index_generator(num_doc, target), request_size=2048)
+
+
+def index_documents(num_docs_index, targets):
+
+    f = Flow.load_config('flows/index.yml')
+
+    with f:
+        f.index(index_generator(num_docs_index, targets), request_size=2048)
+
+
+def query(num_doc, target: dict):
+    f = Flow.load_config('flows/query.yml')
+
+    with f:
+        f.search(query_generator(num_doc, target), shuffle=False, size=128,
+                 on_done=print_result, request_size=32, top_k=TOP_K)
+
+
+def test_query():
+    targets = {
+        'index-labels': {
+            'url': 'http://fashion-mnist.s3-website.eu-central-1.amazonaws.com/train-labels-idx1-ubyte.gz',
+            'filename': os.path.join('./workspace', 'index-labels')
+        },
+        'query-labels': {
+            'url': 'http://fashion-mnist.s3-website.eu-central-1.amazonaws.com/t10k-labels-idx1-ubyte.gz',
+            'filename': os.path.join('./workspace', 'query-labels')
+        },
+        'index': {
+            'url': 'http://fashion-mnist.s3-website.eu-central-1.amazonaws.com/train-images-idx3-ubyte.gz',
+            'filename': os.path.join('./workspace', 'index')
+        },
+        'query': {
+            'url': 'http://fashion-mnist.s3-website.eu-central-1.amazonaws.com/t10k-images-idx3-ubyte.gz',
+            'filename': os.path.join('./workspace', 'query')
+        }
+    }
+
+    config('index')
+    download_data(targets, None)
+    index_documents(NUM_DOCS_INDEX, targets)
+
+    config('query')
+    query(NUM_DOCS_QUERY, targets)
+    result_set = set(result)
+    contain_query = True if 'Pullover' in result_set else False
```

Comment:
we should check that every single result is a `Pullover`

Reply:
Like using a for loop? I think the test now is equal to the 'for loop check' because I will also check if the size of result_set is 1, which means that all of the results are the same and it is 'Pullover'